scripts/k8s-etcd

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+# [This file is part of the zabbix-agent-osso package]
+# REQUIRES: kubectl
+#
+#UserParameter=k8s-etcd.discovery, sudo /etc/zabbix/scripts/k8s-etcd --discover
+#UserParameter=k8s-etcd.values-status[*], sudo /etc/zabbix/scripts/k8s-etcd --get-values-etcd-status "$1" "$2" "$3" "$4"
+#UserParameter=k8s-etcd.values-health[*], sudo /etc/zabbix/scripts/k8s-etcd --get-values-etcd-health "$1" "$2" "$3" "$4"
+#
+
+get_contexts() {
+    kubectl config get-contexts -oname
+}
+
+get_namespaces() {
+    kubectl get namespace -l ossobv/zabbix-agent-osso.k8s-etcd=true \
+            --context "$1" -oname 2>/dev/null |
+        sed -e 's/namespace\///'
+}
+
+if [ "$1" = "--discover" ]; then
+    for context in $(get_contexts); do
+        for namespace in $(get_namespaces "$context"); do
+            status_name=$(kubectl --context $context get namespace $namespace \
+                -ojsonpath='{.metadata.labels}' |
+                jq -r '."ossobv/zabbix-agent-osso.k8s-etcd-status-name"')
+            endpoints=$(kubectl --context $context  -n $namespace get cm \ 
+                $status_name -ojsonpath='{.data.status}' \
+                | jq -r '.etcd_health[]|.endpoint')
+            for endpoint in $endpoints; do
+                echo '{'"{#CONTEXT}":"'$context'",' \
+                    '"{#NAMESPACE}":"'$namespace'",' \
+                    '"{#STATUS_NAME}":"'$status_name'",' \
+                    '"{#ENDPOINT}":"'$endpoint'"'}'
+            done
+        done
+    done | jq -sc
+    exit 0
+
+elif [ "$1" = "--get-values-etcd-status" ]; then
+    kubectl --context $2  -n $3 get cm $4 -ojsonpath='{.data.status}'\
+     | jq -r '.etcd_status[] | select(.Endpoint ==  "'"$5"'")'
+    exit 0
+
+elif [ "$1" = "--get-values-etcd-health" ]; then
```

Comment:
`health-json`

Reply:
Op zich zou het ook een keuze kunnen zijn om de twee JSON blobs tegelijk op te halen en die samen te voegen tot een json. Dan zit je wel met het naming-probleem. Maar `status[...]` lijkt mij prima ook als daar health-info bij ingedrukt zit.

Reply:
lijkt me iets voor later gooi het op de backlog ( nu geen zin in )